distbench_engine.cc

```diff
@@ -783,13 +824,49 @@ void DistBenchEngine::ActionListState::CancelActivities() {
   int size = action_list->proto.action_names_size();
   for (int i = 0; i < size; ++i) {
     auto action_state = &state_table[i];
-    if (action_state->action->proto.has_activity_name()) {
+    if (action_state->action->proto.has_activity_config_name()) {
       action_state->all_done_callback();
       action_state->finished = true;
     }
   }
 }
 
+// Updates the activities_log_ map with the activity metrics for the activities
+// in current run list. In case two activities have same ActivityConfig, the
+// metrics from these activities are summed up into one metric.
+// For example:
+// The 'iteration_count' for two activities 'A1' and 'A2' that have the same
+// activity config 'AC' and have run 20 and 30 times respectively is reported as
+// iteration_count = 50.
+void DistBenchEngine::ActionListState::UpdateActivitiesLog(
+    std::map<std::string, ActivityLog>* activities_logs) {
+  int size = action_list->proto.action_names_size();
```

Comment:
generic 'size' is not helping us here either. This could just be a range for loop can't it?